virtual_ecosystem/models/soil/model_config.py

```diff
@@ -1,418 +1,401 @@
+    :cite:t:`lin_modelling_2000`. """
 
     primary_phosphorus_breakdown_rate: float = 1.0 / 4.38e6
     """Rate constant for breakdown of primary phosphorus to labile phosphorus [day^-1].
-    
-    Default value taken from :cite:t:`parton_dynamics_1988`.
-    """
+    Default value taken from :cite:t:`parton_dynamics_1988`."""
 
     secondary_phosphorus_breakdown_rate: float = 1.0 / 13500
     """Rate constant for breakdown of secondary mineral to labile phosphorus [day^-1].
-    
-    Default value taken from :cite:t:`parton_dynamics_1988`.
-    """
+    Default value taken from :cite:t:`parton_dynamics_1988`."""
 
     labile_phosphorus_sorption_rate: float = 1.0 / 600
-    """Rate constant for sorption of labile phosphorus to secondary mineral phosphorus.
-    
-    Units of [day^-1]. Default value taken from :cite:t:`parton_dynamics_1988`.
-    """
+    """Rate constant for sorption of labile phosphorus to secondary mineral phosphorus
+    [day^-1]. Default value taken from :cite:t:`parton_dynamics_1988`. """
 
     ammonium_deposition_rate: float = 1.5e-4 / 365.25
-    """Rate at which ammonium is deposited into the system [kg N m^-2 day^-1].
-    
-    We are assuming that deposition rates won't vary substantially over the area the
-    simulation encompasses. Value taken from :cite:t:`vet_global_2014`.
-    """
+    """Rate at which ammonium is deposited into the system [kg N m^-2 day^-1]. We are
+    assuming that deposition rates won't vary substantially over the area the 
+    simulation encompasses. Value taken from :cite:t:`vet_global_2014`."""
 
     phosphorus_deposition_rate: float = 5e-6 / 365.25
     """Rate at which phosphorus is deposited into the system [kg P m^-2 day^-1].
-    
     We are assuming that deposition rates won't vary substantially over the area the
-    simulation encompasses. Value taken from :cite:t:`Mahowald2008`.
-    """
+    simulation encompasses. Value taken from :cite:t:`Mahowald2008`."""
 
     nitrogen_fixer_supply_fraction: float = 0.3
     """Fraction of carbon supplied by plants to symbiotes that goes to nitrogen fixers.
-    
-    [unitless]. The remainder goes to mycorrhizal fungi.
-    """
+    [unitless]. The remainder goes to mycorrhizal fungi."""
 
     ectomycorrhiza_supply_fraction: float = 0.25
     """Fraction of plant carbon supply to mycorrhizal fungi that goes to ectomycorrhiza.
-    
-    [unitless]. The remainder goes to arbuscular mycorrhizal fungi.
-    """
+    [unitless]. The remainder goes to arbuscular mycorrhizal fungi."""
+
+    @model_validator(mode="after")
+    def _check_pH_sequence(cls) -> SoilConstants:
+        """Checks that the microbe pH values are in the correct order."""
+        if not (
+            cls.min_pH_microbes
+            < cls.lowest_optimal_pH_microbes
+            < cls.highest_optimal_pH_microbes
+            < cls.max_pH_microbes
+        ):
+            raise ValueError("Microbe pH values not in increasing sequence")
```

Comment:
Probably slightly easier to understand if these are called "thresholds" rather than "values" (apologies if this is just copied directly from something I previously wrote 🫠)